httpcore/parallel.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+import typing
+from types import TracebackType
+
+
+class Parallel:
```

Comment:
I think I'd like to see an interface more like `concurrent.futures.Executor`

eg
```python
with httpx.parallel.Parallel() as p:
    pr = p.submit(url)
    pr2 = p.submit(url2)
    for response in httpx.parallel.as_completed(pr, pr2, timeout=x):
        ...

    # map is a bit of an odd name though
    for response in p.map(httpx.Request(url), httpx.Requst(url2)):
        ...
```

Another feature I'd like is to have an iterable of Requests and get back a MappingProxyType of Request -> Response